### ISSUE TITLE
Not working with special characters

### ISSUE THREAD

**Issue by meltuhamy** [olivernn/lunr.js#275]
Lunr doesn't seem to work with special characters. Here are some examples

https://jsfiddle.net/00y9o4b7/2/
```js
var lunrIdx = lunr(function() {
  this.field('title');
  this.add({title: "مرحبا"}); 
  this.add({title: "你好"});
  this.add({title: "¿dónde?"});
  this.add({title: "why?"});
  this.add({title: "(bracket)"});
 });
 
alert(lunrIdx.search('مرحبا').length); // 0
alert(lunrIdx.search('你好').length); // 0
alert(lunrIdx.search('¿dónde?').length); // 0
alert(lunrIdx.search('why?').length); // 0
alert(lunrIdx.search('(bracket)').length); // 0
```

**Comment by olivernn** [comment]
There are a couple of things at play here, I'll try and explain each one...

1. "(bracket)" [`lunr.trimmer`](https://github.com/olivernn/lunr.js/blob/master/lib/trimmer.js) is by default present in the indexing pipeline and will remove any leading and trailing non word characters. So you should be able to see that "bracket" is in the index and searchable.

2. "why?" is affected by both the `lunr.trimmer` and the [`lunr.stopWordFilter`](https://github.com/olivernn/lunr.js/blob/master/lib/stop_word_filter.js). First the "?" is removed, after which the token "why" is a stop word so is not indexed.

3. "مرحبا" and "你好" are again artefacts of the `lunr.trimmer`. It uses a regex to find non word characters that unfortunately does not play nicely with non-latin languages. If you need to support languages other than English, and especially if they are non-latin based, you should take a look at the excellent [lunr-languages](https://github.com/MihaiValentin/lunr-languages) project.

4. "¿dónde?" at first I thought this was a similar issue to 3, but it is more like the first issue with "(bracket)". The `lunr.trimmer` removes the question marks and leaves just `dónde` in the index.

So, in summary, if you need to search for non latin (or non English even) languages you will get the best results from using the lunr-languages plugin. For the other cases, the pipeline is customisable, so you can _remove_ the trimmer if you want, or add it in the search pipeline (though that might strip any wildcards from a search term). The stopWordFilter is also customisable and you can generate one based on a set of words that make sense for your corpus, either that or completely remove it.

**Comment by olivernn** [comment]
Closing this as I don't think there is anything more to do, feel free to re-open or comment if you disagree.
